launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ContextDataManager.java

```diff
@@ -152,8 +125,12 @@ private void initDataInternal(
         EnvironmentData oldData;
         ContextIndex newIndex;
 
-        synchronized (writerLock) {
-            currentContext = context;
+        synchronized (lock) {
+            if (!context.equals(currentContext)) {
+                // if incoming new data is not for the current context, reject it.
+                return;
+            }
```

Comment:
For reviewers:  I considered refactoring this code to be willing to store the data for the non-current context if such a situation was to occur, but it seemed unlikely and the refactor would be without risk.  There is also an interesting edge case / race condition if that non-current context was not already cached and it put the cache over its cacheLimit.